eli5/xgboost.py

```diff
@@ -1,21 +1,41 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
+import re
 from singledispatch import singledispatch
+from typing import Any, Dict, List, Tuple
 
 import numpy as np
-from xgboost import XGBClassifier, XGBRegressor
+import scipy.sparse as sp
+from xgboost import XGBClassifier, XGBRegressor, Booster, DMatrix
 
-from eli5.base import FeatureWeight, FeatureImportances, Explanation
-from eli5.explain import explain_weights
-from eli5.sklearn.utils import get_feature_names
-from eli5.utils import argsort_k_largest_positive
+from eli5.base import (
+    FeatureWeight, FeatureImportances, Explanation, TargetExplanation)
+from eli5.formatters.features import FormattedFeatureName
+from eli5.explain import explain_weights, explain_prediction
+from eli5.sklearn.text import add_weighted_spans
+from eli5.sklearn.utils import (
+    get_feature_names, get_X, handle_vec, predict_proba)
+from eli5.utils import argsort_k_largest_positive, get_target_display_names
+from eli5._feature_weights import get_top_features
 
 
 DESCRIPTION_XGBOOST = """
 XGBoost feature importances; values are numbers 0 <= x <= 1;
 all values sum to 1.
 """
 
+if not hasattr(XGBRegressor, 'feature_importances_'):
+    # xgboost <= 0.6a2, fixed in https://github.com/dmlc/xgboost/pull/1591
+
+    def xgb_feature_importances(xgb):
+        b = xgb.booster()
+        fs = b.get_fscore()
+        all_features = np.array(
+            [fs.get(f, 0.) for f in b.feature_names], dtype=np.float32)
+        return all_features / all_features.sum()
+
+    XGBRegressor.feature_importances_ = property(xgb_feature_importances)
```

Comment:
hm, what do you think about making it an utility function which gets feature importances from XGBRegressor, instead of patching XGBRegressor directly?